src/swap/hooks/useSwapBalances.tsx

```diff
@@ -48,5 +57,13 @@ export function useSwapBalances({
 
     toBalanceString,
     toTokenBalanceError,
+
+    refetchBalances: async () => {
```

Comment:
I believe we miss tests for this.

Reply:
Btw, I don't think this is a clean way to write this. We are creating a callback inside an object, and than we are sending this callback everywhere.

This approach makes debugging this very complex. We need to rewrite this.

We need a way to start the `refetch` when at Provider level by listening to a particular case.